Undo/COUndoTrack.m

```diff
@@ -566,13 +573,30 @@ - (void) doCommand: (COCommandGroup *)aCommand inverse: (BOOL)inverse addToStore
 {
 	COCommandGroup *commandToApply = (inverse ? [aCommand inverse] : aCommand);
 	[commandToApply setParentUndoTrack: self];
-	
+
 	NSMutableDictionary *md = [NSMutableDictionary new];
-	md[kCOCommitMetadataIdentifier] = inverse ? @"org.etoile.CoreObject.undo" : @"org.etoile.CoreObject.redo";
-	if ([aCommand localizedShortDescription] != nil)
+	NSString *identifier = aCommand.metadata[kCOCommitMetadataIdentifier];
+	NSString *shortDescription = aCommand.metadata[kCOCommitMetadataShortDescription];
+	NSArray *args = aCommand.metadata[kCOCommitMetadataShortDescriptionArguments];
+	NSNumber *inversedValue = aCommand.metadata[kCOCommitMetadataNodeInversed];
+	BOOL inversed = inverse && (inversedValue == nil || !inversedValue.boolValue);
```

Comment:
I changed the name of the metadata key to kCOCommitMetadataUndoInitialBaseInversed.

Having just 'BOOL inversed = inverse' seems useless, since you can infer it easily by checking the kCOCommitMetadataUndoType (when the commit identifier is suffixed by 'undo', inversed is YES)

The inversed value I'm proposing could be computed by the client code, but it's easy to get it wrong and it would force you to access other commands than the current one, since you need to traverse the undo base command chain until you reach the initial base.

I use it to decide whether I applied a strikethrough style to the text. For example, suppose we have:

- Rename
- Selective Undo (Rename)
- Selective Undo (Selective Undo (Rename))

With this dedicated key, it can easily be presented as:

- Rename (kCOCommitMetadataUndoInitialBaseInversed == NO)
- ~~Rename~~ (kCOCommitMetadataUndoInitialBaseInversed == YES)
- Rename (kCOCommitMetadataUndoInitialBaseInversed == NO)